components/camel-huaweicloud-smn/src/main/java/org/apache/camel/component/huaweicloud/smn/SimpleNotificationProducer.java

```diff
@@ -0,0 +1,366 @@
+                default:
+                    if (LOG.isErrorEnabled()) {
+                        LOG.error("Unsupported service name {}", service);
+                    }
+                    throw new UnsupportedOperationException(String.format("service %s is not a supported service", service));
+            }
+        } else {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Service name is null/empty");
+            }
+            throw new IllegalStateException("service name cannot be null/empty");
+        }
+    }
+
+    /**
+     * Publish message service operations
+     *
+     * @param endpoint
+     * @param exchange
+     */
+    private void performPublishMessageServiceOperations(SimpleNotificationEndpoint endpoint, Exchange exchange) {
+        PublishMessageResponse response;
+
+        PublishMessageRequestBody apiBody;
+        this.clientConfigurations = validateServiceConfigurations(endpoint, exchange);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Checking operation name");
+        }
+        switch (clientConfigurations.getOperation()) {
+
+            case SmnOperations.PUBLISH_AS_TEXT_MESSAGE:
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Publishing as text message");
+                }
+                apiBody = new PublishMessageRequestBody()
+                        .withMessage(exchange.getMessage().getBody(String.class))
+                        .withSubject(clientConfigurations.getSubject())
+                        .withTimeToLive(String.valueOf(clientConfigurations.getMessageTtl()));
+
+                response = smnClient.publishMessage(new PublishMessageRequest()
+                        .withBody(apiBody)
+                        .withTopicUrn(clientConfigurations.getTopicUrn()));
+                break;
+
+            case SmnOperations.PUBLISH_AS_TEMPLATED_MESSAGE:
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Publishing as templated message");
+                }
+                apiBody = new PublishMessageRequestBody()
+                        .withMessage(exchange.getMessage().getBody(String.class))
+                        .withSubject(clientConfigurations.getSubject())
+                        .withTimeToLive(String.valueOf(clientConfigurations.getMessageTtl()))
+                        .withMessageTemplateName((String) exchange.getProperty("CamelHwCloudSmnTemplateName"))
+                        .withTags((HashMap<String, String>) exchange.getProperty("CamelHwCloudSmnTemplateTags"))
+                        .withTimeToLive(String.valueOf(clientConfigurations.getMessageTtl()));
+
+                response = smnClient.publishMessage(new PublishMessageRequest()
+                        .withBody(apiBody)
+                        .withTopicUrn(clientConfigurations.getTopicUrn()));
+                break;
+
+            default:
+                throw new UnsupportedOperationException(
```

Comment:
also here a default operation instead of throwing an error

Reply:
I think defaulting to publishAsTextMessage is good idea. i'll check it out

Reply:
> also here a default operation instead of throwing an error

@omarsmak 
After analysis, I suggest it would be nice to keep it concrete with mandatory/right operation name because input for each operations are slightly different. if  user misses out on correct inputs/properties, then API might error out. Sdk team will be expanding support for more services and operations. so once we have a bigger list, we could analyze on a good candidate to become a default service/operation